lib/rules/selector-max-compound-selectors/__tests__/index.mjs

```diff
@@ -384,3 +384,82 @@ testRule({
 		},
 	],
 });
+
+// Testing ignoreSelectors
+testRule({
+	ruleName,
+	config: [2, { ignoreSelectors: ['::v-deep', '/-moz-.*/', /-screen$/] }],
+
+	accept: [
+		{
+			code: '::v-deep .foo .bar {}',
+			description: 'ignored selector at the beginning',
+		},
+		{
+			code: '.foo ::v-deep .bar {}',
+			description: 'ignored selector in the middle',
+		},
+		{
+			code: '.foo ::v-deep > .bar {}',
+			description: 'ignored selector in the middle followed by a combinator',
+		},
+		{
+			code: '.foo::v-deep .bar {}',
+			description: 'ignored selector as pseudo-element',
+		},
+		{
+			code: '.foo::v-deep > .bar {}',
+			description: 'ignored selector as pseudo-element followed by a combinator',
+		},
+		{
+			code: '.foo > ::v-deep .bar {}',
+			description: 'ignored selector preceded by a combinator',
+		},
+		{
+			code: '.foo input::-moz-placeholder {}',
+			description: 'selector ignored by string regex',
+		},
+		{
+			code: '.foo :-webkit-full-screen a {}',
+			description: 'selector ignored by regex',
+		},
+		{
+			code: '.foo::v-deep :-webkit-full-screen a ::-moz-placeholder {}',
+			description: 'multiple ignored selectors',
+		},
+		{
+			code: '.foo { & ::v-deep > .bar {} }',
+			description: 'nested ignored selector',
+		},
```

Comment:
question
major

How should `'.foo .baz .bar.quz {}'` behave for `2` ignore `[".bar", ".quz"]`?
i.e. does it remove one by one from the compound selector or does it skip it if it's not exactly a match?

Reply:
@FloEdelmann ping

Reply:
See https://github.com/stylelint/stylelint/pull/7544#pullrequestreview-1931665796